src/main/java/org/wildfly/security/credential/store/impl/CommandCredentialStore.java

```diff
@@ -0,0 +1,147 @@
+        this.attributes = attributes;
+    }
+
+    /**
+     * Check if credential store service supports modification of its store
+     *
+     * @return {@code true} in case of modification of the store is supported, {@code false} otherwise
+     */
+    @Override
+    public boolean isModifiable() {
+        return false;
+    }
+
+    /**
+     * Check whether credential store service has an entry associated with the given credential alias of specified credential type.
+     *
+     * @param credentialAlias key to check existence
+     * @param credentialType  to check existence in the credential store
+     * @return {@code true} in case key exist in store otherwise {@code false}
+     * @throws CredentialStoreException           when there is a problem with credential store
+     * @throws UnsupportedCredentialTypeException when the credentialType is not supported
+     */
+    @Override
+    public <C extends Credential> boolean exists(String credentialAlias, Class<C> credentialType) throws CredentialStoreException, UnsupportedCredentialTypeException {
+        throw log.methodNotImplemented("exists", getName());
+    }
+
+    /**
+     * Store credential to the credential store service under the given alias. If given alias already contains specific credential type type the credential
+     * replaces older one. <em>Note:</em> {@link CredentialStoreSpi} supports storing of multiple entries (credential types) per alias.
+     * Each must be of different credential type.
+     *
+     * @param credentialAlias to store the credential to the store
+     * @param credential      instance of {@link Credential} to store
+     * @throws CredentialStoreException           when the credential cannot be stored.
+     * @throws UnsupportedCredentialTypeException when the credentialType is not supported
+     */
+    @Override
+    public <C extends Credential> void store(String credentialAlias, C credential) throws CredentialStoreException, UnsupportedCredentialTypeException {
+        throw log.methodNotImplemented("store", getName());
+    }
+
+    /**
+     * Retrieve credential stored in the store under the key and of the credential type
+     *
+     * @param credentialAlias to find the credential in the store
+     * @param credentialType  - credential type to retrieve from under the credentialAlias from the store
+     * @return instance of {@link Credential} stored in the store
+     * @throws CredentialStoreException           - if credentialAlias credentialType combination doesn't exist or credentialAlias cannot be retrieved
+     * @throws UnsupportedCredentialTypeException when the credentialType is not supported
+     */
+    @Override
+    public <C extends Credential> C retrieve(String credentialAlias, Class<C> credentialType) throws CredentialStoreException, UnsupportedCredentialTypeException {
+        Assert.assertNotNull(credentialAlias);
```

Comment:
You probably want checkNotNull here - assertNotNull is no op if assertions are disabled resulting in a NPE below, checkNotNull will always fail on null leading to a consistent error message.